app/controllers/resources_controller.rb

```diff
@@ -161,10 +163,12 @@ def file_sets(model_params)
   end
 
   def signed_ids(model_params)
-    file_sets(model_params).flat_map do |fileset|
-      fileset.dig(:structural, :contains).filter_map do |file|
-        # Only include ActiveStorage signed IDs
-        file[:externalIdentifier] if signed_id?(file[:externalIdentifier])
+    {}.tap do |signed_ids|
+      file_sets(model_params).flat_map do |fileset|
+        fileset.dig(:structural, :contains).filter_map do |file|
+          # Only include ActiveStorage signed IDs
+          signed_ids[file[:filename]] = file[:externalIdentifier] if signed_id?(file[:externalIdentifier])
```

Comment:
We don't currently require it. However, as a practical matter since cocina files correspond with files on disk and uniqueness is enforced by the file system, I don't see this as a significant risk. And if it is, we can enforce in cocina.